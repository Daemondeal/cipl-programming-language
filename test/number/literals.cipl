print(123)
print(968132)
print(0)
print(-0)

print(123.456)
print(-0.001)