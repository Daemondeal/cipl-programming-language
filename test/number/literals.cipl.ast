(expression (call print 123))
(expression (call print 968132))
(expression (call print 0))
(expression (call print (- 0)))
(expression (call print 123.456))
(expression (call print (- 0.001)))

