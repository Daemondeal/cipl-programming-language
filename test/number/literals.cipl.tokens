Token          Lexeme         Line 
IDENTIFIER     print          1    
LEFT_PAREN     (              1    
NUMBER         123            1     123
RIGHT_PAREN    )              1    
LINE_END       \n             2    
IDENTIFIER     print          2    
LEFT_PAREN     (              2    
NUMBER         968132         2     968132
RIGHT_PAREN    )              2    
LINE_END       \n             3    
IDENTIFIER     print          3    
LEFT_PAREN     (              3    
NUMBER         0              3     0
RIGHT_PAREN    )              3    
LINE_END       \n             4    
IDENTIFIER     print          4    
LEFT_PAREN     (              4    
MINUS          -              4    
NUMBER         0              4     0
RIGHT_PAREN    )              4    
LINE_END       \n             5    
IDENTIFIER     print          6    
LEFT_PAREN     (              6    
NUMBER         123.456        6     123.456
RIGHT_PAREN    )              6    
LINE_END       \n             7    
IDENTIFIER     print          7    
LEFT_PAREN     (              7    
MINUS          -              7    
NUMBER         0.001          7     0.001
RIGHT_PAREN    )              7    
LINE_END       \n             7    
EOF                           7    
