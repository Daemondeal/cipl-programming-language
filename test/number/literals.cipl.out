::stdout::
123
968132
0
-0
123.456
-0.001

::stderr::
